src/Providers/Http/HttpTransporter.php

```diff
@@ -66,13 +68,21 @@ public function __construct(
      * {@inheritDoc}
      *
      * @since 0.1.0
+     * @since n.e.x.t Added optional RequestOptions parameter and ClientWithOptions support.
      */
-    public function send(Request $request): Response
+    public function send(Request $request, ?RequestOptions $options = null): Response
     {
         $psr7Request = $this->convertToPsr7Request($request);
+        $options = $options ?? $request->getOptions();
```

Comment:
We shouldn't do a general either or here IMO, but if both are set, we should merge them.

A key decision here is the precedence, and I'm not sure about that:
* Do we prioritize the given parameter because it's more explicit at the time of this call (could be seen as a way to override the regular request)?
* Or do we prioritize the instance within the `Request` DTO because that's defined more closely to the source of truth (the request defines what kind of content is requested, so things like timeout may directly depend on those considerations)?

Curious what you think.

Either way, I also think we should document the decision above this line here with a brief inline comment.